cmd/podman/shared/events.go

```diff
@@ -0,0 +1,115 @@
+package shared
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/containers/libpod/libpod/events"
+	"github.com/containers/libpod/pkg/util"
+	"github.com/pkg/errors"
+)
+
+func generateEventFilter(filter, filterValue string) (func(e *events.Event) bool, error) {
+	switch strings.ToUpper(filter) {
+	case "CONTAINER":
+		return func(e *events.Event) bool {
+			if e.Type != events.Container {
+				return false
+			}
+			if e.Name == filterValue {
+				return true
+			}
+			return strings.HasPrefix(e.ID, filterValue)
+		}, nil
+	case "EVENT", "STATUS":
+		return func(e *events.Event) bool {
+			return fmt.Sprintf("%s", e.Status) == filterValue
+		}, nil
+	case "IMAGE":
+		return func(e *events.Event) bool {
+			if e.Type != events.Image {
+				return false
+			}
+			if e.Name == filterValue {
+				return true
+			}
+			return strings.HasPrefix(e.ID, filterValue)
+		}, nil
+	case "POD":
+		return func(e *events.Event) bool {
+			if e.Type != events.Pod {
+				return false
+			}
+			if e.Name == filterValue {
+				return true
+			}
+			return strings.HasPrefix(e.ID, filterValue)
+		}, nil
+	case "VOLUME":
+		return func(e *events.Event) bool {
+			if e.Type != events.Volume {
+				return false
+			}
+			return strings.HasPrefix(e.ID, filterValue)
```

Comment:
just checking, do you need an `if e.name == filtervalue {` statement here for Volumes?

Reply:
volumes dont have names

Reply:
No, volumes only have names, not IDs